packages/components/src/components/tabs/tabs.jsx

```diff
@@ -24,34 +25,7 @@ const Tabs = ({
     const active_tab_ref = React.useRef();
     const tabs_wrapper_ref = React.useRef();
 
-    React.useEffect(() => {
-        let initial_index = active_index;
-        if (should_update_hash) {
-            const hash = location.hash.slice(1);
-            const hash_index = children.findIndex(child => child.props && child.props['data-hash'] === hash);
-            if (hash_index !== -1) {
-                initial_index = hash_index;
-                history.push({ hash });
-            }
-        }
-        onClickTabItem(initial_index || 0);
-        setActiveLineStyle();
-    }, [active_tab_index, location.hash]);
-
-    const onClickTabItem = index => {
-        if (should_update_hash) {
-            const hash = children[index].props['data-hash'];
-            if (hash && hash !== location.hash.slice(1)) {
-                history.push({ hash });
-            }
-        }
-        if (typeof onTabItemClick === 'function') {
-            onTabItemClick(index);
-        }
-        setActiveTabIndex(index);
-    };
-
-    const setActiveLineStyle = () => {
+    const setActiveLineStyle = React.useCallback(() => {
```

Comment:
Since we are adding `setActiveLineStyle` as a dependency at [Line #73](https://github.com/binary-com/deriv-app/pull/2533/files#diff-7d8d67f5298fd376f554e2544f42bdf5f9a452fe48624da39db112d642ceaed1R73), we need to use `useCallback` here. Otherwise, `useEffect` at [Line #73](https://github.com/binary-com/deriv-app/pull/2533/files#diff-7d8d67f5298fd376f554e2544f42bdf5f9a452fe48624da39db112d642ceaed1R73) would run every time

Reply:
If you're sure that a function doesn't change, you can safely ignore that dependency in the dependency array, but I get what you mean.